libtrap/src/ifc_file.c

```diff
@@ -150,9 +177,8 @@ char *create_filename_from_time(void *priv)
       return NULL;
    }
 
-   strncpy(new_filename, config->filename, config->filename_base_length);
-   
-   strftime(new_filename + config->filename_base_length, 14, ".%Y%m%d%H%M", localtime(&config->starting_time));
+   strftime(new_filename, config->filename_base_length + 6, config->filename, tm_info);
```

Comment:
I have a strong suspicion this approach will not work.
1) on line 175 you must allocate more memory
2) I might be mistaken, but config->filename no longer has formating string, does it?

Did you actually test this approach?